docs/projects/project-setup.md

```diff
@@ -0,0 +1,217 @@
+```
+.
+├── .env
+└── src
+    ├── functions
+    │   ├── data.py
+    │   └── train.py
+    ├── project_setup.py
+    └── workflows
+        └── main_workflow.py
+```
+
+The `project_setup.py` script looks like the following:
+```python
+import os
+
+import mlrun
+
+
+def setup(project: mlrun.projects.MlrunProject) -> mlrun.projects.MlrunProject:
+    source = project.get_param("source")
+    secrets_file = project.get_param("secrets_file")
+    default_image = project.get_param("default_image")
+
+    # Set project git/archive source and enable pulling latest code at runtime
+    if source:
+        print(f"Project Source: {source}")
+        project.set_source(project.get_param("source"), pull_at_runtime=True)
+
+    # Create project secrets and also load secrets in local environment
+    if secrets_file and os.path.exists(secrets_file):
+        project.set_secrets(file_path=secrets_file)
+        mlrun.set_env_from_file(secrets_file)
+
+    # Set default project docker image - functions that do not specify image will use this
+    if default_image:
+        project.set_default_image(default_image)
+
+    # MLRun Functions - note paths are relative to project context (./src)
+    project.set_function(
+        name="get-data",
+        func="functions/data.py",
+        kind="job",
+        handler="get_data",
+    )
+
+    project.set_function(
+        name="train",
+        func="functions/train.py",
+        kind="job",
+        handler="train_model",
+    )
+
+    # MLRun Workflows - note paths are relative to project context (./src)
+    project.set_workflow("main", "workflows/main_workflow.py")
+
+    # Save and return the project:
+    project.save()
+    return project
+```
+
+The project can then be loaded using the following code snippet:
+```python
+project = mlrun.get_or_create_project(
+    name="my-project",
+    context="./src", # project_setup.py should be in this directory
+    parameters={
+        "source" : "https://github.com/mlrun/my-repo#main",
+        "secrets_file" : ".env",
+        "default_image" : "mlrun/mlrun"
+    }
+)
+```
+
+## Common Operations
+
+Some common operations that can be added to the `project_setup.py` script include:
+
+
+### Set Project Source
+
+Set the project source and enable pulling at runtime if specified. See {py:meth}`~mlrun.projects.MlrunProject.set_source` for more info.
+
+```python
+source = project.get_param("source") # https://github.com/mlrun/my-repo#main
+
+project.set_source(source, pull_at_runtime=True)
+```
+
+### Export Project to Zip File Archive
+Export local project directory contents to a zip file archive. Use this in conjunction with setting the project source for rapid iteration without requiring a Git commit for each change. See {py:meth}`~mlrun.projects.MlrunProject.set_source` and {py:meth}`~mlrun.projects.MlrunProject.export` for more info.
+
+***Note:** This requires using the Iguazio `v3io` data layer or some `s3` compliant object storage such as `minio`.*
```

Comment:
**Note:** This requires using the Iguazio `v3io` data layer or some `s3` compliant object storage such as `minio`.